0.13/hoarder/src/main/scala/org/romanowski/hoarder/actions/CachedCI.scala

```diff
@@ -1,22 +1,24 @@
 package org.romanowski.hoarder.actions
 
-import java.nio.file.{Files, Path, Paths}
+import java.nio.file.Path
+import java.nio.file.Paths
 
 import org.romanowski.HoarderSettings._
 import org.romanowski.hoarder.core.HoarderEngine
 import sbt.Def._
 import sbt.Keys._
 import sbt._
 
-object CachedCI {
+object CachedCI extends HoarderEngine {
 
   trait Setup {
```

Comment:
Not sure what you've got set up, but I noticed that you mention Travis. Ironically, I'm using Drone and Appveyor for my testing environments. I may add Travis back later.

What I need is the ability to stash the artefacts for a specific build and store them somewhere that I can reuse later. I need to keep separate my cache for Appveyor, Drone, and (maybe in the future) Travis, since I would be using each as a different build environment (Windows, Mac, Linux). Even if I should be able to compile once and run anywhere, I would still prefer to keep my cache created on Windows separate for that created on Linux.

I was looking at https://github.com/minio/minio, which matches the s3 storage API. The thought was to use its Java client, which is simpler and (I think) more permissible than the Amazon library. Furthermore, I could run my own s3 instance via minio on my server and keep my binary artefacts there.